extensions/arc/src/ui/dashboards/postgres/postgresResourceHealthPage.ts

```diff
@@ -0,0 +1,343 @@
+				});
+			});
+
+			if (p.role.toUpperCase() !== loc.coordinator.toUpperCase()) {
+				podNames.push(p.name);
+			} else {
+				this.coordinatorData = podHealthModels;
+				this.coordinatorPodName = p.name;
+				podNames.unshift(p.name);
+			}
+			this.podConditionsTableIndexes.set(p.name, indexes);
+		});
+
+		this.podConditionsTable.data = this.podsData.map(p => [p.condition, p.details, p.lastUpdate]);
+
+		return podNames;
+	}
+
+	private findPodIssues(): void {
+		const podStatus = this._postgresModel.config?.status.podsStatus;
+		let issueCount = 0;
+
+		podStatus?.forEach(p => {
+			p.conditions.forEach(c => {
+				if (c.status === 'False') {
+					issueCount++;
+				}
+			});
+
+			if (issueCount > 0) {
+				this.podIssuesDetected.push(loc.numberOfIssuesDetected(p.name, issueCount));
+				issueCount = 0;
+			}
+		});
+	}
+
+	private refreshPodSummarySection(): void {
+		let podSummaryTitle = this.modelView.modelBuilder.flexContainer().withProps({
+			CSSStyles: { 'alignItems': 'center', 'height': '15px', 'margin-top': '20px' }
+		}).component();
+		if (!this._postgresModel.config) {
+			podSummaryTitle.addItem(this.modelView.modelBuilder.loadingComponent().component(), { CSSStyles: { 'margin-right': '5px' } });
+			podSummaryTitle.addItem(this.modelView.modelBuilder.text().withProps({
+				value: loc.loading,
+				CSSStyles: { ...cssStyles.title }
+			}).component());
+			this.podSummaryContainer.addItem(podSummaryTitle);
+		} else {
+			let components: azdata.Component[] = [];
+			let imageComponent = this.modelView.modelBuilder.image().withProps({
+				iconPath: IconPathHelper.success,
+				width: iconSize,
+				height: iconSize,
+				iconHeight: '20px',
+				iconWidth: '20px'
+			}).component();
+
+			this.findPodIssues();
+			if (this.podIssuesDetected.length === 0) {
+				imageComponent.iconPath = IconPathHelper.success;
+				podSummaryTitle.addItem(imageComponent, { CSSStyles: { 'margin-right': '5px' } });
+				podSummaryTitle.addItem(this.modelView.modelBuilder.text().withProps({
+					value: loc.available,
+					CSSStyles: { ...cssStyles.title, 'margin-left': '0px' }
+				}).component());
+				components.push(podSummaryTitle);
+				components.push(this.modelView.modelBuilder.text().withProps({
+					value: loc.noPodIssuesDetected,
+					CSSStyles: { ...cssStyles.text, 'margin-top': '20px' }
+				}).component());
+			} else {
+				imageComponent.iconPath = IconPathHelper.fail;
+				podSummaryTitle.addItem(imageComponent, { CSSStyles: { 'margin-right': '5px' } });
+				podSummaryTitle.addItem(this.modelView.modelBuilder.text().withProps({
+					value: loc.issuesDetected,
+					CSSStyles: { ...cssStyles.title }
+				}).component());
+				components.push(podSummaryTitle);
+				components.push(this.modelView.modelBuilder.text().withProps({
+					value: loc.podIssuesDetected,
+					CSSStyles: { ...cssStyles.text, 'margin-top': '20px 0px 10px 0px' }
+				}).component());
+				this.podIssuesDetected.forEach(i => {
```

Comment:
I find it clearer for cases like this to do something like `components.push(...this.podIssuesDetected.map(i => ....`